akka-docs-dev/rst/scala/http/routing-dsl/directives/index.rst

```diff
@@ -0,0 +1,224 @@
+.. _Directives:
+
+Directives
+==========
+
+A "Directive" is a small building block used for creating arbitrarily complex :ref:`route structures <Routes>`.
+Akka HTTP already pre-defines a large number of directives and you can easily construct your own:
+
+.. toctree::
+   :maxdepth: 1
+
+   alphabetically
+   by-trait
+   custom-directives
+
+
+Basics
+------
+
+Directives create :ref:`Routes`. To understand how directives work it is helpful to contrast them with the "primitive"
+way of creating routes.
+
+Since ``Route`` is just a type alias for a function type ``Route`` instances can be written in any way in which function
+instances can be written, e.g. as a function literal::
+
+    val route: Route = { ctx => ctx.complete("yeah") }
```

Comment:
these snippets are not compiled, so they won't react to changes in API or refactorings in IDE.